io_scene_godot/converters/grease.py

```diff
@@ -0,0 +1,367 @@
+
+
+    def generate_surfaces(self, escn_file, export_settings, mesh):
+        """Splits up the mesh into surfaces with a single material each.
+        Within this, it creates the Vertex structure to contain all data about
+        a single vertex
+        """
+        surfaces = []
+        for glayer in mesh.layers:
+            if not len(glayer.frames) or not len(glayer.frames[0].strokes):
+                continue
+
+            for stroke in glayer.frames[0].strokes:
+
+                # Find a surface that matches the material, otherwise create a new
+                # surface for it
+                surface_index = self.mesh_resource.get_surface_id(
+                    stroke.material_index
+                )
+                if surface_index is None:
+                    surface_index = len(surfaces)
+                    self.mesh_resource.set_surface_id(
+                        stroke.material_index, surface_index
+                    )
+                    surface = Surface()
+                    surface.id = surface_index
+                    surfaces.append(surface)
+                    if mesh.materials:
+                        if stroke.material_index >= len(mesh.materials):
+                            print('WARN: bad face.material_index')
+                            mat = mesh.materials[-1]
+                        else:
+                            mat = mesh.materials[stroke.material_index]
+                        if (mat is not None and
+                                export_settings['use_export_material']):
+                            surface.material = export_material(
+                                escn_file,
+                                export_settings,
+                                self.object,
+                                mat
+                            )
+
+                surface = surfaces[surface_index]
+                vertex_indices = []
+
+                for vert_id in range(len(stroke.points)):
+                    vert = stroke.points[vert_id]
+                    new_vert = Vertex( vert )
+                    surface.vertex_data.vertices.append(new_vert)
+                    surface.vertex_data.indices.append(vert_id)
+
+                    # Merge similar vertices
+                    #tup = new_vert.get_tup()
+                    #if tup not in surface.vertex_map:
+                    #    surface.vertex_map[tup] = len(surface.vertex_data.vertices)
+                    #    surface.vertex_data.vertices.append(new_vert)
+
+                    #vertex_index = surface.vertex_map[tup]
+                    #surface.vertex_index_map[vert_id] = vertex_index
+                    #vertex_indices.append(vertex_index)
+
+                #if len(vertex_indices) > 2:  # Only triangles and above
+                #    surface.vertex_data.indices.append(vertex_indices)
+
+        for surface in surfaces:
+            self.mesh_resource[surface.name_str] = surface
+
+
+class VerticesArrays:
```

Comment:
Not have much knowledge of Grease Object, so is these tagent, UV, etc all necessary?

Reply:
i'm also new to grease pencil, so i'm not sure if it has tangents or UV's

Reply:
i wish there was a way to detect blender prims, but there's no extra metadata on them.
one workaround is we provide our own buttons for making prims, which just call the blender prim ops, but we add the extra `gdprim` and `gdprim_radius` custom attributes automatically to the prim.

Reply:
how the gd* attributes autogen a gdscript is complex, and probably better if i explain it a wiki, or other doc.  If you setup a wiki for this project, i will add a page there explaining how this works.

Reply:
> i wish there was a way to detect blender prims, but there's no extra metadata on them.

Yeah, looks like there isn't any. Maybe use the custom attributes is the best thing we can do.

> how the gd* attributes autogen a gdscript is complex, and probably better if i explain it a wiki, or other doc. If you setup a wiki for this project, i will add a page there explaining how this works.

Create a wiki sounds very good. Unfortunately the wiki of github repo is not forkable, you can probably start writing the basic idea of your design in the wiki of your own [fork](https://github.com/hartsantler/godot-blender-exporter) just for the purpose of reviewing this PR ^ ^. 
And I can help to create a full document in [Godot Doc](https://docs.godotengine.org/en/3.1/getting_started/workflow/assets/escn_exporter/index.html?highlight=exporter) once the gdscript feature merged in.